pkg/codegen/dotnet/gen.go

```diff
@@ -534,7 +537,7 @@ func (pt *plainType) genInputPropertyAttribute(w io.Writer, indent string, prop
 	if prop.IsRequired() {
 		attributeArgs = ", required: true"
 	}
-	if pt.res != nil && pt.res.IsProvider {
+	if pt.res != nil && pt.res.IsProvider && !pt.mod.doNotJSONEncodeProviderConfiguration {
```

Comment:
Same concern here as with other languages, is pt.mod looking up the flag from the right context or it needs to be more nuanced for foreign provider references?